tm2/pkg/bft/rpc/lib/client/http/client.go

```diff
@@ -237,7 +225,18 @@ func parseRemoteAddr(remoteAddr string) (network string, s string, err error) {
 	case 2:
 		protocol, address = parts[0], parts[1]
 	}
-	return protocol, address, nil
+
+	// Append default ports if not specified
+	if !strings.Contains(address, ":") {
+		switch protocol {
+		case protoHTTPS:
+			address += ":443"
+		default:
+			address += ":80" // HTTP, TCP
```

Comment:
```suggestion
		case protoHTTP, protoTCP:
			address += ":80"
		default: // noop
```

I think it's more explicit and future-proof like this.